pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -73,16 +73,30 @@ func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *mo
 		// IP allocation logic for service entry was unable to allocate an IP.
 		if svcAddress == constants.UnspecifiedIP {
 			// For all k8s headless services, populate the dns table with the endpoint IPs as k8s does.
-			// TODO: Need to have an entry per pod hostname of stateful set but for this, we need to parse
-			// the stateful set object, associate the object with the appropriate kubernetes headless service
-			// and then derive the stable network identities.
+			// And for each individual pod, populate the dns table with the endpoint IP with a manufactured host name from
+			// "statefulset.kubernetes.io/pod-name" that k8s sets.
 			if svc.Attributes.ServiceRegistry == string(serviceregistry.Kubernetes) &&
 				svc.Resolution == model.Passthrough && len(svc.Ports) > 0 {
 				// TODO: this is used in two places now. Needs to be cached as part of the headless service
 				// object to avoid the costly lookup in the registry code
 				for _, instance := range push.ServiceInstancesByPort(svc, svc.Ports[0].Port, nil) {
 					// TODO: should we skip the node's own IP like we do in listener?
 					addressList = append(addressList, instance.Endpoint.Address)
+					if pod, ok := instance.Endpoint.Labels["statefulset.kubernetes.io/pod-name"]; ok {
+						address := []string{instance.Endpoint.Address}
+						host := pod + "." + svc.Attributes.Name + "." + svc.Attributes.Namespace + "." + node.DNSDomain // Follow k8s name convention.
```

Comment:
Shouldn't it be the DNS domain the pod is actually running in? Not sure we can achieve that though...

Also, I thought `node.DNSDomain` was like `namespace.svc.cluster.local`? so we have namespace 2x?

Reply:
yeah. I may be I will verify that if DNS domain has namespace and if yes, I will exclude it